trieste/acquisition/rule.py

```diff
@@ -67,13 +70,15 @@ class AcquisitionRule(ABC, Generic[T_co, SP_contra]):
     (see e.g. :class:`TrustRegion`). Indeed, to use an :class:`AcquisitionRule` in the main
     :class:`~trieste.bayesian_optimizer.BayesianOptimizer` Bayesian optimization loop, the rule
     must return either a `TensorType` or `State`-ful `TensorType`.
+
+    An :class:`AcquisitionRule` can only be passed models of type :class:`M_contra`.
     """
 
     @abstractmethod
     def acquire(
         self,
         search_space: SP_contra,
-        models: Mapping[str, ProbabilisticModel],
+        models: Mapping[str, M_contra],
```

Comment:
previous name was more descriptive, as a user it would be easier to understand - you can go to check the class etc, here we have additional step to figure out what is this M_contra